### ISSUE TITLE
Periodically run integration tests with valgrind

### ISSUE THREAD

**Issue by engelmi** [engelmi/hirte#325]
## Please describe what you would like to see

In order to detect memory leaks, the integration tests should be run via [valgrind](https://valgrind.org/). However, since running valgrind on top will slow down the test execution, running those enhanced tests periodically should be sufficient. 

## Please describe the solution you'd like

- adding valgrind to the [integration-test-base](https://github.com/containers/hirte/blob/main/tests/containers/integration-test-base) image
- adding an option to the existing [HirteTest class](https://github.com/containers/hirte/blob/main/tests/hirte_test/test.py#L14-L21) to wrap the `ExecStart` of the systemd service for `hirte` and `hirte-agent` with valgrind
   - flag can be set via env variable (thus passed in from tmt) and provided via fixture
   - based on the flag, the systemd service needs to be changed before starting it
   - valgrind should log its output to file so that it can be gathered and added to the artifacts in the CI
- a new job to run the integration tests with valgrind along with the regular integration tests

## Please describe your use case

In order to detect memory leaks, the integration tests should be run via [valgrind](https://valgrind.org/).

**Comment by dougsland** [comment]
+1

**Comment by rhatdan** [comment]
I like the idea, but who will monitor this?  If the test do not cause a failure on a PR, then users will not notice.

**Comment by engelmi** [comment]
> I like the idea, but who will monitor this? If the test do not cause a failure on a PR, then users will not notice.

Good question to which I, unfortunately, don't have a good answer. 
I was thinking about creating a github action badge that references the latest run with valgrind (green = its fine, red = fails) and then some of the hirte maintainers needs to investigate... but thats rather fluffy and no quality gate.

**Comment by mwperina** [comment]
If it's possible to send an email about this valgrind GH workflow failure to all maintainers of the project, then it should be enough.

https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/notifications-for-workflow-runs

**Comment by dougsland** [comment]
> If it's possible to send an email about this valgrind GH workflow failure to all maintainers of the project, then it should be enough.
> 
> https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/notifications-for-workflow-runs

+1

**Comment by dougsland** [comment]
@engelmi I was thinking here, CI don't cover this task already? Every new PR it trigger valgrind check correct?

**Comment by engelmi** [comment]
> @engelmi I was thinking here, CI don't cover this task already? Every new PR it trigger valgrind check correct?

Yes and no. For the unit tests we already do this (in the CI we just trigger `make test-with-valgrind`). For the integration tests we don't, its also a bit more complicated then. (sorry for the late reply, I missed this question)

**Comment by mkemel** [comment]
Now we run integration test on each push.
I think running integration tests with valgrind would be beneficial for each PR, or at least for each PR that changes .c/.h files in the project. So maybe it should be mandatory to run it before merging (like ost in oVirt).
What do you think?

**Comment by dougsland** [comment]
> Now we run integration test on each push.
> I think running integration tests with valgrind would be beneficial for each PR, or at least for each PR that changes .c/.h files in the project. So maybe it should be mandatory to run it before merging (like ost in oVirt).
> What do you think?

agreed.

**Comment by mkemel** [comment]
As discussed in the bluechi weekly meeting, from what we've seen - running integration tests with valgrind does not take that much longer than running without it, so we might as well run them in this configuration each time along with the regular one. Updated the issue description

**Comment by mkemel** [comment]
With #641 merged, the work on this issue is concluded.
